tasks.rb

```diff
@@ -9,37 +9,43 @@
 def create_hpxmls
   this_dir = File.dirname(__FILE__)
   workflow_dir = File.join(this_dir, 'workflow')
-  hpxml_inputs_tsv_path = File.join(workflow_dir, 'hpxml_inputs.tsv')
+  hpxml_inputs_tsv_path = File.join(workflow_dir, 'hpxml_inputs.json')
 
-  require 'csv'
-  csv_data = CSV.open(hpxml_inputs_tsv_path, headers: :first_row, col_sep: "\t").map(&:to_h)
+  require 'json'
+  json_inputs = JSON.parse(File.read(hpxml_inputs_tsv_path))
   abs_hpxml_files = []
-  dirs = csv_data.map { |r| r['hpxml_path'] }.uniq
+  dirs = json_inputs.keys.map { |file_path| File.dirname(file_path) }.uniq
 
   schema_path = File.join(File.dirname(__FILE__), 'HPXMLtoOpenStudio', 'resources', 'hpxml_schema', 'HPXML.xsd')
   schema_validator = XMLValidator.get_schema_validator(schema_path)
 
-  puts "Generating #{csv_data.size} HPXML files..."
+  puts "Generating #{json_inputs.size} HPXML files..."
 
-  csv_data.each_with_index do |csv_row, i|
-    puts "[#{i + 1}/#{csv_data.size}] Generating #{csv_row['hpxml_file']}..."
-    hpxml_file = csv_row['hpxml_file']
-    csv_row['hpxml_path'] = File.join(workflow_dir, csv_row['hpxml_path'], hpxml_file)
-    hpxml_path = csv_row['hpxml_path']
+  json_inputs.keys.each_with_index do |hpxml_filename, i|
+    puts "[#{i + 1}/#{json_inputs.size}] Generating #{hpxml_filename}..."
+    hpxml_path = File.join(workflow_dir, hpxml_filename)
     abs_hpxml_files << File.absolute_path(hpxml_path)
-    csv_row.delete('hpxml_file')
 
-    # Convert from true/false strings to boolean
-    csv_row.each do |k, v|
-      next if v.nil?
+    # Build up json_input from parent_json(s)
+    parent_json_filenames = []
+    parent_json_filename = json_inputs[hpxml_filename]['parent_json']
+    while not parent_json_filename.nil?
+      if not json_inputs.keys.include? parent_json_filename
+        fail "Could not find parent_json: #{parent_json_filename}."
+      end
 
-      csv_row[k] = true if ['true', 'TRUE'].include?(v)
-      csv_row[k] = false if ['false', 'FALSE'].include?(v)
+      parent_json_filenames << parent_json_filename
+      parent_json_filename = json_inputs[parent_json_filename]['parent_json']
+    end
+    json_input = { 'hpxml_path' => hpxml_path }
+    for parent_json_filename in parent_json_filenames.reverse
+      json_input.merge!(json_inputs[parent_json_filename])
     end
+    json_input.merge!(json_inputs[hpxml_filename])
+    json_input.delete('parent_json')
 
     measures = {}
-    measures['BuildResidentialHPXML'] = [csv_row]
-    # measures['BuildResidentialScheduleFile'] = [sch_args] if !sch_args.empty?
```

Comment:
We no longer call `BuildResidentialScheduleFile` to re-generate csv files in `HPXMLtoOpenStudio/resources/schedule_files/`? Do we need to ensure these files are kept up-to-date?

Reply:
Good question. I think I lost that capability by accident. Let me see if I can somewhat easily bring it back.

Reply:
Addressed in last commit.